includes/sanitizers/class-amp-style-sanitizer.php

```diff
@@ -399,6 +418,7 @@ public function __construct( DOMDocument $dom, array $args = [] ) {
 		$this->base_url    = untrailingslashit( $guessurl );
 		$this->content_url = WP_CONTENT_URL;
 		$this->xpath       = new DOMXPath( $dom );
+		$this->cache_pool  = new AMP_Cache_Pool( self::CACHE_GROUP );
```

Comment:
This is not being injected because I don't think the architecture is currently set up well for injecting dependencies.
For now, I wanted to keep it simple and just directly instantiated here. Hopefully, we'll have injection for v2.

Reply:
Ah, OK. I thought this was dependency injection.

Reply:
Dependency injection would be if exterior code would inject "some cache" and this code here would not need to know which exact implementation is being used. This would then decouple this code from the cache, and the cahe could be changed/replaced without needing to adapt this code here.

Reply:
OK, thanks!